cookie-manager.js

```diff
@@ -0,0 +1,228 @@
+"use strict";
+
+(function () {
+
+    const init = function () {
+        readConfiguration();
+    };
+
+    const readConfiguration = function() {
+
+        const xhr = new XMLHttpRequest();
+        xhr.open('GET', '/cookie-manager.json');
+
+        xhr.onreadystatechange = function() {
+            if (xhr.readyState !== 4) {
+                return;
+            }
+            if (xhr.status !== 200) {
+                console.error('HTTP Error whilst requesting "/cookie-manager.json".', xhr.status, xhr.statusText);
+                scriptError('Unable to continue without a valid configuration file.');
+                return;
+            }
+
+            let configurationJson;
+            try {
+                configurationJson = JSON.parse(xhr.responseText);
+            } catch (e) {
+                console.error('Unable to parse /cookie-manager.json as JSON', e);
+                scriptError('Unable to continue without a valid configuration file.');
+            }
+
+            configurationLoaded(configurationJson);
+        };
+
+        xhr.send();
+    };
+
+    const configurationLoaded = function (configuration) {
+        manageCookies(configuration);
+        findAndBindPreferencesForm(configuration);
+    };
+
+    const manageCookies = function(configuration) {
+
+        const cm_cookie = configuration['user-preference-cookie-name'];
+        const cm_user_preferences = getUserPreferences(cm_cookie);
+
+        if (typeof cm_user_preferences === 'undefined') {
+            console.info(
+                'User preference cookie is not set or valid. ' +
+                'This cookie defines user preferences. ' +
+                'Assuming non-consent, and deleting all non-essential cookies.'
```

Comment:
Non-essential cookies are only deleted if the relevant configuration property is set :)